Google.Api.Gax.Grpc/AsyncResponseStream.cs

```diff
@@ -0,0 +1,79 @@
+    ///   <item>Pre-C# 8 users who will be calling <see cref="MoveNextAsync()" /> and <see cref="Current"/> directly don't need
+    ///   to call <see cref="GetAsyncEnumerator(CancellationToken)"/>.</item>
+    /// </list>
+    /// Note that <see cref="GetAsyncEnumerator(CancellationToken)"/> can only be called once per instance due to the "only iterate once"
+    /// nature of the response stream.
+    /// </remarks>
+    /// <typeparam name="TResponse">The response type.</typeparam>
+    public sealed class AsyncResponseStream<TResponse> : IAsyncEnumerable<TResponse>, IAsyncEnumerator<TResponse>
+    {
+        private readonly IAsyncStreamReader<TResponse> _reader;
+        private int _getEnumeratorCalled;
+        private CancellationToken _cancellationToken;
+
+        internal AsyncResponseStream(IAsyncStreamReader<TResponse> reader) =>
+            _reader = reader;
+
+        /// <inheritdoc />
+        public TResponse Current => _reader.Current;
+
+        // The gRPC response stream doesn't support disposal. The gRPC call (AsyncDuplexStreamingCall or AsyncServerStreamingCall) does,
+        // but users should dispose that themselves if they want to rather than us doing it here.
+        /// <inheritdoc />
+        public ValueTask DisposeAsync() => default;
+
+        /// <summary>
+        /// Begins iterating over the response stream, using the specified cancellation token. This method can only be called
+        /// once per instance.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token to use in subsequent <see cref="MoveNextAsync()"/> calls.</param>
+        /// <exception cref="InvalidOperationException">This method has already been called on this instance.</exception>
+        /// <returns>An iterator over the response stream.</returns>
+        public IAsyncEnumerator<TResponse> GetAsyncEnumerator(CancellationToken cancellationToken = default)
+        {
+            if (Interlocked.CompareExchange(ref _getEnumeratorCalled, 0, 1) != 0)
+            {
+                throw new InvalidOperationException($"{nameof(GetAsyncEnumerator)} can only be called once for a gRPC response stream wrapper.");
+            }
+            _cancellationToken = cancellationToken;
+            return this;
+        }
+
+        /// <summary>
+        /// Moves to the next item, using the specified cancellation token.
+        /// </summary>
+        /// <remarks></remarks>
+        /// <param name="cancellationToken">The cancellation token to use for this step.</param>
+        /// <returns>A task that will complete with a result of true if the enumerator was successfully advanced to the next element, or false if the enumerator has passed the end of the collection.</returns>
+        public async ValueTask<bool> MoveNextAsync(CancellationToken cancellationToken) =>
+            await _reader.MoveNext(_cancellationToken).ConfigureAwait(false);
```

Comment:
`cancellationToken` here I think, not `_cancellationToken`?

Reply:
Humbug, yes! Will fix now.